extensions/wikia/Insights/InsightsHooks.class.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+class InsightsHooks {
+
+	/**
+	 * Check if article is in insights flow and init script to show banner with message and next steps
+	 */
+	public static function onBeforePageDisplay( OutputPage &$out, Skin &$skin ) {
+		global $wgRequest;
+
+		$insightCategory = $wgRequest->getVal('insights', null);
+
+		if ( $insightCategory && InsightsHelper::isInsightPage( $insightCategory ) ) {
+			$out->addScriptFile( '/extensions/wikia/Insights/scripts/LoopNotification.js' );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Add insight param to keep information about flow after edit
+	 */
+	public static function onArticleUpdateBeforeRedirect( $article, &$sectionanchor, &$extraQuery ) {
+		global $wgRequest;
+
+		$insightCategory = $wgRequest->getVal('insights', null);
+
+		if ( $insightCategory && InsightsHelper::isInsightPage( $insightCategory ) ) {
+			if ( !empty( $extraQuery ) ) {
+				$extraQuery .= '&';
+			}
+			$extraQuery .= 'insights=' . $insightCategory;
```

Comment:
What about ```"insights={$insightsCategory}"```? And I think that we need to decide on wording for consistency - subpage or category?